src/hypertable_cache.c

```diff
@@ -137,6 +138,34 @@ ts_hypertable_cache_get_entry(Cache *cache, Oid relid)
 	return ts_hypertable_cache_get_entry_with_table(cache, relid, NULL, NULL);
 }
 
+/*
+ * Returns hypertable by checking the given cache first.
+ * If hypertable is not found, fails with an error.
+ */
+Hypertable *
+ts_hypertable_cache_get_entry_always(const Oid relid, Cache *const cache)
+{
+	Hypertable *const ht = ts_hypertable_cache_get_entry(cache, relid);
+
+	if (ht == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_TS_HYPERTABLE_NOT_EXIST),
+				 errmsg("table \"%s\" is not a hypertable", get_rel_name(relid))));
+
+	return ht;
+}
+
+/*
+ * Returns cache into the argument and hypertable as the function result.
+ * If hypertable is not found, fails with an error.
+ */
+Hypertable *
+ts_hypertable_get_cache_always(const Oid relid, Cache **const cache)
```

Comment:
Same here w.r.t. naming.

Might also want to allow passing a NULL `cache` argument here and not get the cache (without failure). The cache is automatically released at the end of the transaction, although it is good practice to release it early, if possible. Still, I think having the option to not manage the cache would be good.

Reply:
@erimatnor I am not sure how long the cache is needed. Looking to all the code, which uses hypertable, it keeps cache pinned while hypertable is used and releases the cache only at the end of functions.
So I don't see possible use cases where cache will not be needed if hypertable is returned.